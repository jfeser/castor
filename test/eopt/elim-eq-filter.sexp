((queries
  "filter(a = param0, atuple([ascalar(0 as a), ascalar(0 as b)], cross))"
  "depjoin(ascalar(1 as x), filter(0.x = a, atuple([ascalar(0 as a), ascalar(0 as b)], cross)))")
 (transforms hoist-filter elim-eq-filter))
