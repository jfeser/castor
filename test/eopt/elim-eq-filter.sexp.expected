$3:
  filter((a = param0), atuple([ascalar(0 as a), ascalar(0 as b)], cross))
  ahashidx(dedup(select([a], atuple([ascalar(0 as a), ascalar(0 as b)], cross))), $7, param0)
  
$4:
  filter((0.x = a), atuple([ascalar(0 as a), ascalar(0 as b)], cross))
  ahashidx(dedup(select([a], atuple([ascalar(0 as a), ascalar(0 as b)], cross))), $7, 0.x)
  
$6:
  depjoin(ascalar(1 as x), $4)
  
$7:
  filter((a = 0.a), atuple([ascalar(0 as a), ascalar(0 as b)], cross))
  ahashidx(dedup(select([a], atuple([ascalar(0 as a), ascalar(0 as b)], cross))), $7, 0.a)
  
